From 023d6a4039c05bdac17288653ba0095b90635213 Mon Sep 17 00:00:00 2001
From: Brendan King <Brendan.King@imgtec.com>
Date: Thu, 11 Apr 2024 08:06:43 +0100
Subject: [PATCH 65/65] egl/wayland: allow sharing of protected buffers

Allow protected buffers to be shared between compatible GPUs.

When the IMG Mesa patches are ported to the next major version of
Mesa, this change should be merged with commit "dri2: add a driver
compatibility check extension".
---
 src/egl/drivers/dri2/platform_wayland.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/egl/drivers/dri2/platform_wayland.c b/src/egl/drivers/dri2/platform_wayland.c
index 8359653b17f..f1fb5b3655a 100644
--- a/src/egl/drivers/dri2/platform_wayland.c
+++ b/src/egl/drivers/dri2/platform_wayland.c
@@ -1248,8 +1248,8 @@ get_back_bo(struct dri2_egl_surface *dri2_surf, bool allow_preserve,
    use_flags = __DRI_IMAGE_USE_SHARE | __DRI_IMAGE_USE_BACKBUFFER;
 
    if (dri2_surf->base.ProtectedContent) {
-      /* Protected buffers can't be read from another GPU */
-      if (dri2_dpy->fd_render_gpu != dri2_dpy->fd_display_gpu)
+      /* Protected buffers can't be read from an incompatible GPU */
+      if (!dri2_dpy->compat_gpus)
          return -1;
       use_flags |= __DRI_IMAGE_USE_PROTECTED;
    }
-- 
2.34.1

